src/main/java/org/apache/mesos/hdfs/Scheduler.java

```diff
@@ -284,17 +283,20 @@ private boolean launchNode(SchedulerDriver driver, Offer offer,
           .build();
       tasks.add(task);
 
-      liveState.addStagingTask(task.getTaskId(), taskName);
-      persistentState.addHdfsNode(taskId, offer.getHostname(), taskType);
+      liveState.addStagingTask(task.getTaskId());
+      persistentState.addHdfsNode(taskId, offer.getHostname(), taskType, taskName);
     }
     driver.launchTasks(Arrays.asList(offer.getId()), tasks);
     return true;
   }
 
   private String getNextTaskType(String taskType) {
+    Collection<String> nameNodeTaskNames = persistentState.getNameNodeTaskNames().values();
+    Collection<String> journalNodeTaskNames = persistentState.getJournalNodeTaskNames().values();
+
     if (taskType.equals(HDFSConstants.NAME_NODE_ID)) {
       for (int i = 1; i <= HDFSConstants.TOTAL_NAME_NODES; i++) {
-        if (!liveState.getNameNodeNames().containsValue(HDFSConstants.NAME_NODE_ID + i)) {
+        if (!nameNodeTaskNames.contains(HDFSConstants.NAME_NODE_ID + i)) {
           return HDFSConstants.NAME_NODE_ID + i;
```

Comment:
Since we're never supposed to reach here when we've launched enough tasks, lets return an error instead of ""